pkg/resources/sku/sku.go

```diff
@@ -190,28 +182,60 @@ func (p AProductConfig) Configure(obj metav1.Object) error {
 		checkResourceBlock(&t.Spec.KeycloakDeploymentSpec.Resources)
 		p.mutateResources(t.Spec.KeycloakDeploymentSpec.Resources.Requests, resources.Requests)
 		p.mutateResources(t.Spec.KeycloakDeploymentSpec.Resources.Limits, resources.Limits)
-		return nil
+		break
+	case *threescalev1.APIManager:
+		checkApiManager(t)
+
+		p.mutateAPIManagerReplicas(t.Spec.Apicast.ProductionSpec.Replicas, ApicastProductionName)
+		p.mutateResourcesRequirement(t.Spec.Apicast.ProductionSpec.Resources, ApicastProductionName)
+
+		p.mutateAPIManagerReplicas(t.Spec.Backend.ListenerSpec.Replicas, BackendListenerName)
+		p.mutateResourcesRequirement(t.Spec.Backend.ListenerSpec.Resources, BackendListenerName)
+
+		p.mutateAPIManagerReplicas(t.Spec.Backend.WorkerSpec.Replicas, BackendWorkerName)
+		p.mutateResourcesRequirement(t.Spec.Backend.WorkerSpec.Resources, BackendWorkerName)
+
 	default:
 		return errors.New(fmt.Sprintf("sku configuration can only be applied to Deployments, StatefulSets or Deployment Configs, found %s", reflect.TypeOf(obj)))
 	}
 
-	if p.sku.isUpdated || *replicas < configReplicas {
+	return nil
+}
+
+func (p AProductConfig) mutateAPIManagerReplicas(replicas *int64, name string) {
+	configReplicas := p.resourceConfigs[name].Replicas
+	value := int64(configReplicas)
+	if p.sku.isUpdated || *replicas < value || *replicas == 0 {
+		*replicas = value
+	}
+}
+
+func (p AProductConfig) mutatePodTemplate(template *corev1.PodTemplateSpec, name string) {
+	for i, _ := range template.Spec.Containers {
+		p.mutateResourcesRequirement(&template.Spec.Containers[i].Resources, name)
+	}
+}
+
+func (p AProductConfig) mutateReplicas(replicas *int32, name string) {
+	if replicas == nil {
+		replicas = &[]int32{0}[0]
```

Comment:
@sergioifg94 I think I could use help with reviewing this bit of code and wonder if you could share your opinion on it?

Reply:
There's a couple of things I'm not sure about in this function.

1. When the value of `replicas` is `nil`, we're only assigning the local variable in the function, but not the value that is passed to it, so if we try the following, the value of `myReplicas` will still be `nil`:
    ```go
    var myReplicas *int32 = nil
   productConfig.mutateReplicas(myReplicas, name)

   fmt.Println(myReplicas) // prints <nil>
    ```

    Is there any test that verifies this? If it's ok to leave the value unchanged when it's `nil`, it'd be better to simply `return` in the `if` block
2. When assigning the value of `replicas`, Is there a reason why we take the address of the first element of a newly created slice instead of using `new(int32)`?